relay-dynamic-config/src/metrics.rs

```diff
@@ -4,11 +4,40 @@ use std::collections::BTreeSet;
 
 use relay_base_schema::data_category::DataCategory;
 use relay_common::glob2::LazyGlob;
+use relay_common::glob3::GlobPatterns;
+use relay_metrics::Bucket;
 use relay_protocol::RuleCondition;
 use serde::{Deserialize, Serialize};
 
 use crate::project::ProjectConfig;
 
+/// Configuration for metrics filtering.
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+#[serde(transparent)]
+pub struct Metrics {
+    /// Patterns of names of metrics that we want to filter.
+    pub denied_names: GlobPatterns,
+}
+
+impl Metrics {
+    /// Filters metrics based on the metric name.
+    pub fn filter_metrics(&self, metrics: &mut Vec<Bucket>) {
+        metrics.retain(|bucket| !self.denied_names.is_match(&bucket.name));
+    }
+
+    /// Creates a new instance of [`Metrics`].
+    pub fn new(patterns: Vec<String>) -> Self {
```

Comment:
_nit_: As this struct will get more fields, we will probably not need this constructor and instead construct it via:

```rust
Metrics {
  denied_names: vec![],
  ..Metrics::default()
}
```